### DISCUSSION TITLE
Questions on sourcing & programming

### DISCUSSION THREAD

**Discussion by Drizzt321** [oxplot/fpx#25] [Q&A]
So a few questions. I recently came across this as I'm contemplating a project which I'd like to charge via USB-PD (and then a buck-boost CC/CV if I can find one to fit inside, otherwise entire charger is external box), and while I'd like to use 20v@3a, not all USB-PD source support it, so I'd like it to gracefully negotiate downward to the max voltage & amp that the source will support. 

On your Configurator, I notice it only allows for a primary & secondary voltage/amp, rather than max preference and scale down to a minimum. 

Also, where can I buy some assembled & flashed ones? I'd REALLY love to get ahold of at least 1, as I'm pretty sure I can fit it, although it'd be tight.

**Comment by oxplot** [comment]
I don't know of any other USB-PD trigger module that's as small as fpx and since I've archived this project (and no longer have any stock left), unfortunately I don't have any suggestions.

But if you can compromise on the size, you can find some good boards [on Tindie](https://www.tindie.com/search/?q=usb-pd).

**Reply by Drizzt321** [reply]
So you're not making any more? Drat. I suppose I could use JLCPCB from the files in your github probably. And you have some decent instructions on flashing, I probably could figure it out.

Size is somewhat of an issue, and the few I've found on Tindie that does similar to what yours does, the shops are "on a break". *sigh*

**Comment by bjkayani** [comment]
I did make a PD board that does what you want. You can set the maximum
voltage and it will go up to that or get as close as it can. Voltage is set
with a solder jumper array and there is no fancy current control or
negotiation.

Here it is compared to the fpx pcb.

If you are interested, I can share the design files or sell you the
assembled boards at a reasonable price.

Let me know.

Regards

**Reply by Drizzt321** [reply]
@bjkayani where's the info/size/etc? You didn't link to anywhere.

**Comment by bjkayani** [comment]
It uses the CH224K IC
<https://www.lcsc.com/product-detail/USB-ICs_WCH-Jiangsu-Qin-Heng-CH224K_C970725.html>
and
is 21x14.5mm.

**Reply by Drizzt321** [reply]
That's just the IC, was looking for something to a Tindie store or something with the rest of the details on functionality.

I actually found https://www.jacobsparts.com/items/POWTRIG-B-20V, which seems actually just what I want. Start at 20v@5a negotiation, and work your way down the voltage levels. And it's quite small too, even smaller than the FPX.

**Comment by bjkayani** [comment]
Oh nice. Yeah there are plenty of options from AliExpress that would do
what you want.

Glad to hear you found a solution.

Cheers

**Comment by oxplot** [comment]
> That's just the IC, was looking for something to a Tindie store or something with the rest of the details on functionality.
> 
> I actually found https://www.jacobsparts.com/items/POWTRIG-B-20V, which seems actually just what I want. Start at 20v@5a negotiation, and work your way down the voltage levels. And it's quite small too, even smaller than the FPX.

Sure, it's smaller but how do you securely mount it? :wink:

**Reply by Drizzt321** [reply]
Some clever 3D printed structure and E6000 glue ;)

But yeah, a point to a degree. Although I think I'm going to change directions a bit, not sure I need something quite that small, since I can't seem to find a decent buck-boost CC/CV that'll handle ~50-60W I want to be able to do, that's small enough to fit inside ~1.25" diameter tube. 

I did find https://www.tindie.com/products/arkelectronics/bq25703a-programmable-regulator-module/, which actually seems like it'd fit the bill, but I think it's juuuuust slightly too large in one dimension (https://github.com/AlexKlimaj/BQ25703A_Module). 

So for now, I'll just use a larger external buck-boost CC/CV, and have 12v anderson inputs, or the USB-PD input, and a long enough cable and a XT30 connector to handle the ~3-3.4a max I'll be doing.

Maybe eventually I'll try and duplicate the board, slightly tweaking the dimensions, and adding an onboard or external ESP8266 (arduino lib https://github.com/Lawmate/BQ25703A) or something like that, to handle configuring the charger and tweaking down the charge current based on if the input is <20V USB-C, or if it's the XT30, assume the source is capable of handling the ~50-60W. Also probably monitoring the temps of the BQ25703A, cut down the charge current if it starts to heat up too much. 

So a future project. First let me get the parts I know I can do together, handle that, and then go ahead and look at something rather advanced like that.
